invenio_circulation/transitions/transitions.py

```diff
@@ -192,6 +194,14 @@ def after(self, loan):
         loan['end_date'] = loan['end_date'].isoformat()
         super(ItemAtDeskToItemOnLoan, self).after(loan)
 
+    def ensure_item_is_available(self, loan):
+        """Validate if item is available for checkout."""
+        if not is_item_waiting_at_desk(loan['item_pid']):
```

Comment:
if the transitions are correctly configured, the only way this transition can be triggered is from status `ITEM_AT_DESK` to `ITEM_ON_LOAN`, otherwise there is a config issue.
Isn't it enough to check with `is_item_available`? This will ensure that in the meantime the loan has not taken by someone else.
Is there any specific need to check for this?

Reply:
for what I was thinking, this transition is a bit different, because there are two points of view. One for every other transition, when we basically check if item is available which means it's not in one of CIRCULATION_STATES_LOAN_ACTIVE. But in the transition from ITEM_AT_DESK to ITEM_ON_LOAN it's different, because we need to check if the item is waiting at desk, ready for checkout, so if it is in ITEM_AT_DESK state, which is considered at not available for others but available for checkout

Reply:
Well, the main point is that you should trust the previous transitions, otherwise anything can fail.
The only thing to do here it is probably just to remove `self.ensure_item_is_available(loan)`.